src/libstd/sys/windows/fs2.rs

```diff
@@ -111,8 +120,31 @@ impl DirEntry {
     }
 
     pub fn path(&self) -> PathBuf {
+        self.root.join(&self.file_name())
+    }
+
+    pub fn file_name(&self) -> OsString {
```

Comment:
I'm not really fine with that. I find it not clear. It's really a shame that I was aware of the RFC so lately...